tapestry.js

```diff
@@ -975,41 +1005,102 @@ function getBoundedCoord(coord, maxCoord) {
     return Math.max(MAX_RADIUS, Math.min(maxCoord - MAX_RADIUS, coord));
 }
 
-function getChildren(id) {
-    var children = [];
-    var dataLinks = dataset.links;
-    for (var linkId in dataLinks) {
-        var link = dataLinks[linkId];
-        //SEARCH FOR: CURRENT_ID_NODE -> CHILD links
-        if (typeof link.source === 'number' && link.source === id) {
-            children.push(link.target);
-        } else if (typeof link.source === 'object' && link.source.id === id) {
-            children.push(link.target.id);
-        }
+// add 'depth' parameter recursively to each node
+function addDepth(rootId, depth, visitlist) {
+    var visited = visitlist;
+    visited.push(rootId);
 
-        //SEARCH FOR: PARENT -> CURRENT_ID_NODE links;
-        // These should also appear as "children" because they're adjacent to the current node
-        if (typeof link.target === 'number' && link.target === id) {
-            children.push(link.source);
-        } else if (typeof link.target === 'object' && link.target.id === id) {
-            children.push(link.source.id);
+    var depthAt = 0;
+
+    dataset.nodes[findNodeIndex(rootId)].depth = depth;
+    var children = getChildrenRec(rootId,1);
+
+    var acc;
+
+    while (depthAt < children.length) {
+        for (var childId in children) {
+            if (visited.includes(children[childId])) {
+                acc = depth;
+                if (dataset.nodes[findNodeIndex(children[childId])].depth > acc) {
+                    dataset.nodes[findNodeIndex(children[childId])].depth = acc;
+                }
+                else {
+                    depthAt++;
+                }
+            }
+            else {
+                acc = depth + 1;
+                dataset.nodes[findNodeIndex(children[childId])].depth = acc;
+//                console.log('id: ' + dataset.nodes[findNodeIndex(children[childId])].id + ' depth: ' + dataset.nodes[findNodeIndex(children[childId])].depth)
+                visited.push(children[childId]);
+
+                addDepth(children[childId],acc,visited);
+                depthAt++;
+            }
         }
     }
-
-    return children;
+    
 }
 
-function getGrandchildren(children) {
-    var grandchildren = [];
-    for (var childId in children) {
-        var child = children[childId];
-        var currentGrandchildren = getChildren(child);
-        grandchildren = grandchildren.concat(currentGrandchildren);
+// return the maximum depth from a given rootId
```

Comment:
Use /** notation to include doc comment for your functions please. (See [here](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/#functions) for more info - doesn't need to include all the content mentioned there, but just be wary of how to format your comments)